grails-app/services/org/pih/warehouse/invoice/PrepaymentInvoiceService.groovy

```diff
@@ -432,18 +438,11 @@ class PrepaymentInvoiceService {
         }
     }
 
-    BigDecimal getAmountToInverse(BigDecimal amountInvoiced, BigDecimal amountInverseable) {
-        return Math.abs(amountInvoiced) >= Math.abs(amountInverseable) ? amountInverseable : amountInvoiced
+    BigDecimal getUnitPriceToInverse(BigDecimal unitPriceInvoiced, BigDecimal unitPriceInverseable) {
+        return Math.abs(unitPriceInvoiced) >= Math.abs(unitPriceInverseable) ? unitPriceInverseable : unitPriceInvoiced
     }
 
-    BigDecimal getAmountAvailableToInverse(OrderAdjustment orderAdjustment, InvoiceItem prepaymentItem) {
-        // To determine what is current amount available to inverse we have to add prepayment amount to currently
-        // inversed amount (because both values have different signs). Examples:
-        // prepaymentItem.amount = 10, orderAdjustment.inversedAmount = -3, expected available to inverse = 7
-        // prepaymentItem.amount = -5, orderAdjustment.inversedAmount = 2, expected available to inverse = -3
-        // prepaymentItem.amount = -5, orderAdjustment.inversedAmount = 5, expected available to inverse = 0
-        // The actual inverse item sign will be switched while saving the inverse, because now we only determine
-        // how much of prepayment item's amount is available for inversing.
-        return prepaymentItem.amount + orderAdjustment.inversedAmount
+    BigDecimal getUnitPriceAvailableToInverse(BigDecimal unitPrice, BigDecimal inversedUnitPrice) {
+        return Math.abs(unitPrice) > Math.abs(inversedUnitPrice) ? unitPrice - inversedUnitPrice : 0
```

Comment:
reintroducing this discussion from the other PR: are we allowing a scenario where unit price and inversed unit price have different signs? Like I said previously, that case doesn't really make sense, but if a user gets confused and puts in the wrong sign, we might get into undesirable scenarios here.

Ex: unit price is 1 and inversed unit price is -1. We should return 2 but we return 0.

Maybe because of this we shouldn't allow the signs to differ.

Reply:
> are we allowing a scenario where unit price and inversed unit price have different signs? 

No. **Both** will have the same sign (can be negative or positive but both will have the same, and this is already validated here https://github.com/openboxes/openboxes/blob/release/0.9.2-hotfix1/grails-app/services/org/pih/warehouse/invoice/PrepaymentInvoiceService.groovy#L400). That's why I changed to the unit price from amount, because with the amount it was a wild ride here because amount could have different signs.

Reply:
> are we allowing a scenario where unit price and inversed unit price have different signs? 

No. **Both** will have the same sign (can be negative or positive but both will have the same, and this is already validated here https://github.com/openboxes/openboxes/blob/release/0.9.2-hotfix1/grails-app/services/org/pih/warehouse/invoice/PrepaymentInvoiceService.groovy#L400). That's why I changed to the unit price from amount, because with the amount it was a wild ride here because amount could have different signs.